Update documentation to reflect ConductR standalone mode

fsat opened the pull request.

fsat committed:
Update documentation to reflect ConductR standalone mode

- Installation steps now takes into account separate ConductR Core and ConductR Agent
- HAProxy reload script is now to be supplied as part of the install
- Update Operations Guide to reflect that config files are now specified in conductr.ini and conductr-agent.ini for ConductR Core and ConductR Agent respectively
- Some update Migration Guide re binary incompatibility, and ConductR Core + ConductR Agent changes

huntc commented on src/main/play-doc/general/MigrationGuide.md:
```diff
@@ -1,14 +1,36 @@
+
+## Compulsory bundle roles
+
+ConductR now requires roles to be specified for each node where ConductR bundles run. Prior to installing ConductR consider a topology that has roles representing a DMZ, gateway services, persistance services (databases) and so forth. You can disable role checking by specifying the following in ConductR Core's `conductr.ini`:
```
It is no longer compulsory and is off by default

markusjura replied:
With ConductR 2.0, isn't role checking disabled by default? https://github.com/typesafehub/conductr/blob/master/conductr/src/main/resources/application.conf#L407

huntc commented on src/main/play-doc/general/MigrationGuide.md:
```diff
@@ -1,14 +1,36 @@
+
+ConductR Agent is responsible for running bundle processes. As such, the bundle roles which are previously specified are now to be specified within ConductR Agent.
+
+The ConductR Core and ConductR Agent is available as 2 different packages. The [[installation instructions|Install]] describes the steps of installing ConductR Core and ConductR Agent on the same host. These steps is also applicable should ConductR Core and ConductR Agent installation on different host is desired.
```
are available

markusjura replied:
Why do we say that the installation instructions explain how to install core and agent on the same host and then in the next sentence say, that these instructions are also applicable when installing core and agent on different hosts? This is confusing for me.

Why not saying that it is now possible to run core and agent on the same, but also on different hosts and then refer to the installation instructions?

huntc commented on src/main/play-doc/operation/ClusterTroubleshooting.md:
```diff
@@ -2,7 +2,7 @@
 ## Cluster Split Brain Recovery
 
-Cluster partition is a failure that may occur when running a cluster based application. To recover from this situation ConductR utilizes Akka SBR (Split Brain Recovery). Akka SBR is a part of Akka in the [Typesafe Reactive Platform](http://www.lightbend.com/products/typesafe-reactive-platform) version `15v01p05`.
+Cluster partition is a failure that may occur when running a cluster based application. To recover from this situation ConductR utilizes Akka SBR (Split Brain Recovery). Akka SBR is a part of Akka in the [Typesafe Reactive Platform](http://www.lightbend.com/products/typesafe-reactive-platform) from version `15v01p05`.
```
Should update this with the new RP version for SBR

huntc commented on src/main/play-doc/operation/Install.md:
````diff
@@ -86,29 +91,77 @@ A typical response contains the current members of the cluster (shown here as ju
+By default ConductR Agent and ConductR Core are using `/tmp` as its working directory. Since both ConductR Agent and ConductR Core are installed in the same machine, move the ConductR Agent working directory as such:
+
+```bash
+[172.17.0.1]$ echo -Dconductr.agent.storage-dir=/tmp/conductr-agent/bundles | sudo tee -a /usr/share/conductr-agent/conf/conductr-agent.ini
````
Why don't we just change the agent's default setting so that we then don't have to mention it?

fsat replied:
@huntc - As discussed, will be handled post-alpha:

https://trello.com/c/k5dbGJJ2/283-ensure-core-and-agent-creates-the-required-bundle-directories

markusjura commented on src/main/play-doc/general/MigrationGuide.md:
```diff
@@ -1,14 +1,36 @@
+
+ConductR 2.0 introduces the notion of ConductR Core and ConductR Agent.
+
+ConductR Core holds the state of the cluster, and is responsible for decision making as to which ConductR Agent gets to run which bundles based on the resource available to the said agent. The physical bundle files are stored on the ConductR Core's filesystem and replicated across the cluster.
```
What about: "based on the resource available on the agent."

huntc commented on src/main/play-doc/operation/Install.md:
```diff
@@ -165,45 +243,94 @@ Proxying application endpoints is required when running more than one instance o
+   /etc/init.d/haproxy reload
+else
+  /bin/systemctl reload  haproxy.service
+fi
```
I thought we were doing away with this script and just provide one liners depending on the OS...

huntc commented on src/main/play-doc/operation/Install.md:
````diff
@@ -536,6 +700,31 @@ A typical response contains the current members of the cluster (shown here is a
+```
+
+By default ConductR Agent and ConductR Core are using `/tmp` as its working directory. Since both ConductR Agent and ConductR Core are installed in the same machine, move the ConductR Agent working directory as such:
+
````
Again we can do away with this if we make the did the agent's default

fsat replied:
@huntc - As discussed, will be handled post-alpha:

https://trello.com/c/k5dbGJJ2/283-ensure-core-and-agent-creates-the-required-bundle-directories

huntc commented on src/main/play-doc/operation/ConfigurationRef.md:
```diff
@@ -249,6 +309,53 @@ conductr {
   }
 
+  # Mesos related settings.
+  # There are only used if the 'conductr.mode' is set to 'mesos'
```
Mesos config changed slightly today...

markusjura commented on src/main/play-doc/operation/ConfigurationRef.md:
```diff
@@ -249,6 +309,53 @@ conductr {
+
+      # Resources needed to start a Mesos executor that runs a ConductR agent
+      resources {
+        cpu        = 1.0
```
The resources have changes with the PR https://github.com/typesafehub/conductr/pull/1269.

New values:

```
      resources {
        cpu        = 0.1
        memory     = 268400000
        disk-space = 1000000000
      }
```

fsat committed:
PR feedback

- Remove bundle roles from migration guide as it's still optional
- Minor grammar change on the Migration Guide
- Update mention of SBR version & remove mention of RP
- Update ConductR reference configuration
- Update HAProxy reload script instructions with systemv, upstart, and systemd specific HAProxy reload command

fsat commented on src/main/play-doc/general/MigrationGuide.md:
```diff
@@ -1,18 +1,31 @@
 
-## Roles and topology
 
-ConductR now requires roles to be specified for each node where ConductR bundles run. Prior to installing ConductR consider a topology that has roles representing a DMZ, gateway services, persistance services (databases) and so forth. You can disable role checking by specifying the following in `application.ini`: 
```
@huntc and @markusjura - removed Roles from migration guide as it's still optional by default

fsat commented on src/main/play-doc/general/MigrationGuide.md:
```diff
@@ -1,18 +1,31 @@
+
+ConductR 2.0 introduces the notion of ConductR Core and ConductR Agent.
+
+ConductR Core holds the state of the cluster, and is responsible for decision making as to which ConductR Agent gets to run which bundles based on the resource available on the agent. The physical bundle files are stored on the ConductR Core's filesystem and replicated across the cluster.
```
@markusjura - updated with "based on the resource available on the agent."

fsat commented on src/main/play-doc/operation/ClusterTroubleshooting.md:
```diff
@@ -2,7 +2,7 @@
 ## Cluster Split Brain Recovery
 
-Cluster partition is a failure that may occur when running a cluster based application. To recover from this situation ConductR utilizes Akka SBR (Split Brain Recovery). Akka SBR is a part of Akka in the [Typesafe Reactive Platform](http://www.lightbend.com/products/typesafe-reactive-platform) version `15v01p05`.
+Cluster partition is a failure that may occur when running a cluster based application. To recover from this situation ConductR utilizes Akka SBR (Split Brain Recovery). ConductR uses Akka SBR version `1.0.0` which is a part of Akka's commercial extension.
```
@huntc - update description re Akka SBR version

fsat commented on src/main/play-doc/operation/ConfigurationRef.md:
```diff
@@ -249,6 +309,56 @@ conductr {
+      resources {
+        cpu        = 0.1
+        memory     = 268400000
+        disk-space = 1000000000
```
@huntc and @markusjura - ConductR reference config updated with latest change

fsat commented:
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;dUD62CKZ&#x2F;236-documentation-update-for-version-2-0"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Documentation update for version 2.0</a>
